### ISSUE TITLE
Improving collaboration

### ISSUE THREAD

**Issue by henrinie** [Signbank/FinSL-signbank#210]
Hello NGT fellows @Woseseltops @vanlummelhuizen @ocrasborn

I would like to engage in closer collaboration with you. To begin with I would like to explore the possible ways we could improve communication between each other. If we manage to do that, there is a possibility that we could share some of the work - saving our precious resources.

What kind of communication tools do you use, could I also use them to communicate with you?
Would you be willing to form some kind of group with me on some communication tool/platform?

Some communication tools:
- [Flowdock](https://www.flowdock.com/)
- [Slack](https://slack.com/)
- IRC (internet relay chat), network of your choice

If you use or know some other possible tools for team communication, please let me know.

**Comment by ocrasborn** [comment]
Myself I use none of these, but as Henri and I discussed last week, it would be good to see how we could already move towards working more on shared code, given the low chances that we will get money to do this in a special project.

**Comment by stevecassidy** [comment]
For some other projects I've been using [Gitter](https://gitter.im) recently, it's integrated with Github, you can create a chat room connected to an organisation or a repo.  It works in the browser and there's a mobile app as well.

**Comment by Woseseltops** [comment]
I would be willing to use any of these tools, but this is my experience:
* I always have an IRC client open (there's a channel for all the technical people in our building), so I wouldn't mind adding an extra channel.
* I've used Slack for my other job and liked it.

One thing to keep in mind: what I think these tools are really good at is 'quick communication'; for example, when you are stuck somewhere and want to know if somebody else knows how to solve it by heart. What I think they will less good at, is giving us a good idea of what others are working on, which I think is more likely to lead to code sharing. One approach to give us all more knowledge on what others are working on could be that we agree on a particular tag for our Github issues, like intesting_to_other_signbanks, and that we take on the habit of checking all issues with this tag in other signbanks every 2 weeks or so.

**Comment by henrinie** [comment]
> what I think these tools are really good at is 'quick communication';
That's right. Also I think they have a lower barrier so possibly more communication could happen.

> What I think they will less good at, is giving us a good idea of what others are working on, which I think is more likely to lead to code sharing.
I agree. This is something we might want to discuss about - could we organize and plan the work together?

Gitter, Slack or IRC. All of these seem fine to me, I checked Flowdock's pricing and it doesn't really suit us.
The benefit of these apps is that they handle the logging of chats for you. But on the other hand IRC is better for me since I already use it.

@Woseseltops Could we try and create an IRC chatroom for starters? What networks are you on? I am currently on freenode, IRCNet and Quakenet, but I can join some additional networks too.

**Comment by Woseseltops** [comment]
> Could we try and create an IRC chatroom for starters? 

Okay :).

> I am currently on freenode, IRCNet and Quakenet

I'm at the local university network only. Just give me a server, and I'll join #signbank there on my university days.

**Comment by henrinie** [comment]
Okay, lets try #signbank @ freenode (irc.freenode.net) or the webchat https://webchat.freenode.net/

@Woseseltops could you set up a terminal screen or something so that you could be there 24/7, and eventually see messages sent in at other times than when you are personally online?
If that is not possible, then another solution is needed.

**Comment by Woseseltops** [comment]
Okay, done. I added freenode/signbank to my list of channels I automatically connect to every morning with my IRC client. 

To be able to respond to Henri's wish to also talk to me when I'm not around, I've also set up a screen session with irssi with another account. I'm using https://github.com/alanhamlett/AwayMail-Irssi-Plugin so I get an email message when someone mentions me :) . 

I'm just typing this here so I will be able to find it back; I could have also told about this on IRC :).

**Comment by henrinie** [comment]
So we have created an IRC (internet relay chat) channel on Freenode network (irc.freenode.org).
**#signbank** is the channel name, anyone is welcome to come there.
Currently me, @Woseseltops and @vanlummelhuizen are there.

On this collaboration thing, @stevecassidy what are your plans with auslan-signbank? Are you just going to upgrade the software to work with django1.10 (soon 1.11LTS), or are you going to add new features as well?

**Comment by stevecassidy** [comment]
I’ve just joined the IRC channel but since I don’t use IRC at all I might not be there very regularly.

The work I’m currently doing on Auslan Signbank is to add a few new features:

-          Search and browse by category (eg. Medical, educational, sport)

-          Dialect based search, search for signs in a given region

-          Adding video definitions – the ability to add a video as a definition for a sign

However, prior to starting that I’m doing some refactoring and bringing things up to date following on from the work that my student did last year in a class project.  I’m also porting to Django 1.10 and cleaning up the dependencies as much as I can.

In the last week I’ve refactored the video application to remove all mention of glosses, it now handles tagged videos where the tag is up to the application. The dictionary app then sets the tag as the IDGLOSS rather than having a ForeignKey between the models.  It would be good if this could be shared between the projects – I had a quick look at your video code and there seems to be some additions in FinSL that could perhaps be ported over.

Cheers,

Steve

**Comment by henrinie** [comment]
It is good to hear more about your plans.

> - Dialect based search, search for signs in a given region

If that doesn't exist in auslan, I think it was then implemented in NGT-signbank for the adminview, unless you mean the public view? We don't use the public view at all at the moment, but I believe that NGT-signbank has enabled it. 

> In the last week I’ve refactored the video application to remove all mention of glosses, it now handles tagged videos where the tag is up to the application. The dictionary app then sets the tag as the IDGLOSS rather than having a ForeignKey between the models.

Do you use the TaggedVideo.tag charfield to handle the relationship to Gloss (or it could be what ever, right?)? Do you store the Gloss.idgloss field on it, or Gloss.pk/id (or a combination)?

> It would be good if this could be shared between the projects – I had a quick look at your video code and there seems to be some additions in FinSL that could perhaps be ported over.

The video application is something that definitely could be shared between the projects. We have different requirements though, that need to be taken into account. I think that your solution would work for us for the videos, if the poster images could be stored there as well. I am not sure about NGT-signbank, @ocrasborn @Woseseltops could you please check this? This would be a wonderful opportunity to share code between the projects.

**Comment by henrinie** [comment]
> I think that your solution would work for us for the videos, if the poster images could be stored there as well. 

I have to add one thing here that I forgot about. In our case we would also need to figure out the dataset of the videos.
This is because we have a feature where you can 'bulk upload videos', and there are multiple datasets (and parties working under the same Signbank). We don't want to mix up all the videos, it is much faster to add the relation to a gloss when you don't have to figure out if the video is meant for the dataset you are maintaining.

**Comment by Woseseltops** [comment]
> The video application is something that definitely could be shared between the projects. We have different requirements though, that need to be taken into account. I think that your solution would work for us for the videos, if the poster images could be stored there as well. I am not sure about NGT-signbank, @ocrasborn @Woseseltops could you please check this?

I'm not 100% sure that I understand the question correctly, but what I can say is that we currently have an extremely basic video system with exactly one video per gloss. If the gloss changes names, the video file changes names, if the gloss is deleted, the video files is deleted, etc.

**Comment by henrinie** [comment]
> I'm not 100% sure that I understand the question correctly, but what I can say is that we currently have an extremely basic video system with exactly one video per gloss. If the gloss changes names, the video file changes names, if the gloss is deleted, the video files is deleted, etc.

I think I meant @stevecassidy 's idea generally, and if your requirements could be met with his design?

**Comment by henrinie** [comment]
So it seems that it is possible that all of us could use the same video application.
It is not yet completely clear for me, how in our case we could save Dataset information for the videos, but it can probably be figured out.

I strongly vouch for sharing the video application, because it is something that isn't very complicated, and all of us have pretty similar requirements for it.
New features could be added later, if we just keep the old behaviour intact. With versioning one could keep using older version too.

@stevecassidy what are your thoughts on this? How should we approach this - would you like to do the work yourself, or would you like to share the workload?
Perhaps we need to write down the requirements first?

**Comment by stevecassidy** [comment]
Hi all,
  so I've reached a reasonable point with my modular video application, 
it might be worth your taking a look to see if you can work with it.  
It's in my own repo at the moment:

https://github.com/stevecassidy/signbank-video

I'll move it to the signbank org when I get the chance.  There's a 
sample application to show what it does.  I've basically added two 
labels to the video - category and tag. Since I want to use this for 
gloss videos as well as those in feedback and definitions, I use 
category to denote which of these it is, then the tag to denote the 
IDgloss of the sign (or maybe the id field of the object, still not sure 
about this).

There's no migration path from the existing models yet, I have to work 
on that.

Comments welcome, and perhaps we can move this particular discussion to 
that repo when I've got it set up in the Signbank org.

Steve

**Comment by henrinie** [comment]
Hi Steve,

I will have to check your work at some point.
I would like to check one thing about the **tag** though. If you are planning on using Gloss.idgloss as the tag, there might be issues with characters(?). It is probably not a problem to you, because operate within English language. We need to support two languages: Finnish and Swedish, both of them have characters like äöå, that often cause problems. I am not sure about Dutch, but Chinese might be problematical, I believe both are used in NGT-signbank.

**Comment by stevecassidy** [comment]
That shouldn't be a problem as long as the database can handle utf8 
strings which most can these days. I just tried it on the sqlite based 
test version and it worked fine.

In any case, the other option is just to use the numerical object id of 
the gloss, I'm not sure at the moment whether one is better than the other.

Steve

Henri wrote:

**Comment by henrinie** [comment]
We had problems with python 2 earlier. I believe NGT has also had some problems with python 2.
Might be better with python3 as there is no str()+unicode().

**Comment by stevecassidy** [comment]
Right, as part of updating the codebase I've moved everything to Python 
3 as well.

Steve

Henri wrote:

**Comment by henrinie** [comment]
@stevecassidy Here are some thoughts about the video app and its compatibility to our requirements.
(It would indeed be good to move this discussion to the _signbank-video_ repo).

1. Poster images: We don't generate these on the server, instead our users pick a frame and upload it.
    - I think we would need a FileField for poster images.
    - The extract_frame part would need to be made optional (add a boolean to settings?)
2. Datasets: We need a way to connect uploaded videos that don't have a Gloss set (yet) to a Dataset.
    - I wonder if TaggedVideo could be used like this to solve this issue?
    ```
    category="dataset"
    tag=dataset.pk
    ```
3. We need to be able to upload multiple videos for one Gloss. We use GlossVideo.version to store the defined order of videos to show on gloss page. Users can arrange the videos to a desired order on the gloss page.
   - It seems that TaggedVideoManager would "mess" the version numbers for us if a new video is uploaded. On the other hand, it might not, the uploaded video would just show first and then it could be moved to where the user wants. **Does TaggedVideoManager** only update the version when a new video is added (it seems so)?

- _On a side note:_ I've been working on a feature to make it possible to record videos with a webcam (https://github.com/Signbank/FinSL-signbank/issues/212). It records videos as webm/vp8, and those don't seem to work on all platforms/browsers without additional codecs/software. If we make the move to using __signbank-video__, it would be useful to have the webm/vp8 videos converted to mp4/h264 when uploaded.

There might be some other things that didn't caught my eye.
Hopefully we all can begin using signbank-video, it is a great effort you are making @stevecassidy

**Comment by henrinie** [comment]
Hello everyone,

@ocrasborn @Woseseltops @stevecassidy @vanlummelhuizen
Would you guys be up for a Video conference meeting? (If we can work out a time that suits Australia and Europe).
There are plenty of opportunities for more collaboration due to @stevecassidy working on modularization. Perhaps we should discuss these opportunities and compare our goals, in order to speed things up.

@codev Would you like to take part too? I have no idea what is going on with BSL-signbank, it would be good to hear your (or someone elses) thoughts about this form the perspective of BSL.

**Comment by ocrasborn** [comment]
Will be happy to, once we've finished updating Django/Python, which is our first priority now. So maybe in a week or two?

**Comment by vanlummelhuizen** [comment]
Note that I am on holiday from April 24 until May 6. Back at work on may 8.

**Comment by codev** [comment]
I'd be happy to take part if I'm around though there isn't much to 
report on the BSL side at the moment. But I'd be interested to hear what 
else is going on.

Marc

**Comment by henrinie** [comment]
Would week19 (8 - 12 May) work for you guys?

**Comment by stevecassidy** [comment]
Hi all, sure, would be good to catch up and share notes.  I’d be ok with that week except for Tuesday. Best time seems to be evening here, 8pm is 1pm in Helsinki, 11am in the UK - have I got that right?


Steve

On 11 Apr 2017, 1:51 AM +1000, Henri <notifications@github.com>, wrote:

Would week19 (8 - 12 May) work for you guys?

**Comment by henrinie** [comment]
> Best time seems to be evening here, 8pm is 1pm in Helsinki, 11am in the UK - have I got that right?

It seems to be right.
@stevecassidy what time would suit you best? I believe that in the UK the regular office hours are 9AM-5PM. 9AM GMT+1 would be 6PM GMT+10.

@codev Is 9AM or earlier generally ok for you? Also if there are other people from the BSL team that would like to take part, please invite them.

To recap the **current** timezones (please correct if wrong):
- United Kingdom: **UTC/GMT +1**
- Netherlands: **UTC/GMT +2**
- Finland: **UTC/GMT +3**
- Australia (East) **UTC/GMT +10**

**Comment by codev** [comment]
Anytime after 7am is fine for me, I work at home. I've been in touch 
with Kearsy at UCL who runs the project for her input and she might join 
or send me some notes.

Marc

**Comment by stevecassidy** [comment]
Later is better for me, I’m usually travelling home at 6pm, 8pm is after dinner and I can join in from home.

Steve

On 11 Apr 2017, 7:00 PM +1000, Henri <notifications@github.com>, wrote:

Best time seems to be evening here, 8pm is 1pm in Helsinki, 11am in the UK - have I got that right?

It seems to be right.
@stevecassidy<https://github.com/stevecassidy> what time would suit you best? I believe that in the UK the regular office hours are 9AM-5PM. 9AM GMT+1 would be 6PM GMT+10.

@codev<https://github.com/codev> Is 9AM or earlier generally ok for you? Also if there are other people from the BSL team that would like to take part, please invite them.

To recap the current timezones (please correct if wrong):

  *   United Kingdom: UTC/GMT +1
  *   Netherlands: UTC/GMT +2
  *   Finland: UTC/GMT +3
  *   Australia (East) UTC/GMT +10

**Comment by henrinie** [comment]
Hello, can you check whether this time will work for you? @ocrasborn @Woseseltops @vanlummelhuizen @codev @stevecassidy @juzza81 

Week19, 8th May, 10-12 May **(not Tuesday)** @ 
**8pm (gtm+11)**, Australia East
**1pm (gmt+3)**, Eastern European Summer Time
**12pm (gmt+2)**, Central European Summer Time
**11am (gmt+1)**, British Summer Time

I would expect us to need at least 2 hours for this meeting.
Is Skype okay, or do you prefer something else?

**Comment by ocrasborn** [comment]
Mon 8 or Wed 10 is possible for me

**Comment by codev** [comment]
Should be fine for me.

**Comment by Woseseltops** [comment]
For me May 11 or 12 would be an option (so no overlap with Onno). Maybe we can use doodle.com ?

**Comment by ocrasborn** [comment]
It's more important that you are there, @Woseseltops , so feel free to pick a time without me.

**Comment by henrinie** [comment]
@Woseseltops can you set up a Doodle?
Finding a time seems to have gotten complicated now.

**Comment by henrinie** [comment]
@susanodd would you like to take part too?
Please invite people that might be interested of this, I might not be aware of everyone involved in different Signbank projects.

**Comment by stevecassidy** [comment]
Any of those days is ok with me.

Steve

On 12 Apr 2017, 7:18 PM +1000, Henri <notifications@github.com>, wrote:

Hello, can you check whether this time will work for you? @ocrasborn<https://github.com/ocrasborn> @Woseseltops<https://github.com/Woseseltops> @vanlummelhuizen<https://github.com/vanlummelhuizen> @codev<https://github.com/codev> @stevecassidy<https://github.com/stevecassidy> @juzza81<https://github.com/juzza81>

Week19, 8th May, 10-12 May (not Tuesday) @
8pm (gtm+11), Australia East
1pm (gmt+3), Eastern European Summer Time
12pm (gmt+2), Central European Summer Time
11am (gmt+1), British Summer Time

I would expect us to need at least 2 hours for this meeting.
Is Skype okay, or do you prefer something else?

**Comment by henrinie** [comment]
I created a poll on doodle, please fill in dates that are okay with you.
https://doodle.com/poll/uhe25zrdi5idvfvf

I am not sure if the time zones work correctly, the time is:

> 8pm - 10pm (gtm+11), Australia East 
> 1pm - 3pm (gmt+3), Eastern European Summer Time 
> 12pm - 2pm (gmt+2), Central European Summer Time 
> 11am - 1pm (gmt+1), British Summer Time

**Comment by Woseseltops** [comment]
Filled it out! Sorry for not creating the doodle like you asked, @henrinie (only now catching up with my Github mentions). 

Our options seem to be:
* Monday 8 without Onno and me
* Tuesday 10 without Micha and me
* Thursday 11 without Onno and Marc

Who is most important? :)

**Comment by henrinie** [comment]
I would like to have everyone involved, but it seems pretty tricky now.
Out of the possible dates I would pick Thursday May 11th, leaving Marc and Onno out.
I would really like to have Marc in too. @codev Is there any chance that you could make it May 11th?

We could try another week, but I doubt that it would work any better. Would May 11-19 work better for you guys, especially @ocrasborn ?

**Comment by codev** [comment]
I'm in UCL on another project on the 11th. I haven't got an agenda yet, 
it's possible I could call in but probably won't know until the week 
before. I'm happy to write a few notes on the BSL site if I can't make it.

Marc

**Comment by Woseseltops** [comment]
Sorry, between May 17 and June 8 there are no opportunities for me (various travels). How about adding a few dates to the Doodle a little further into the future? The further away, the larger the chance people are still free.

**Comment by henrinie** [comment]
@stevecassidy @Woseseltops @vanlummelhuizen @codev 
Lets have the meeting on Thursday May 11th at the following time:

> 8pm - 10pm (gtm+11), Australia East
> 1pm - 3pm (gmt+3), Eastern European Summer Time
> 12pm - 2pm (gmt+2), Central European Summer Time
> 11am - 1pm (gmt+1), British Summer Time

Please double check the time since we are on multiple different time zones.
We are going to be using Skype, please make sure that you have it working, and that you have a webcam available. I will email you my Skype user name.

@Woseseltops 
> How about adding a few dates to the Doodle a little further into the future? The further away, the larger the chance people are still free.

I don't think that there is any guarantee that we will find a better time, therefore lets just have the meeting now, as we have already figured out some possible dates. Unfortunately Marc might not make it (lets hope that he can), but perhaps we can have another meeting on a later date.

**Comment by henrinie** [comment]
Hello once again @stevecassidy @Woseseltops @vanlummelhuizen @ocrasborn @codev (please mention others who might be interested in this)!

Would you guys like to have another conference call to discuss about Signbank and how we could share resources? Hopefully we could get everyone together this time.

Would some time in September work for you?

**Comment by stevecassidy** [comment]
Yes, would be good to meet up again. I'd be ok before September 14th.  Similar time constraints as before, most evenings would be ok.

Steve

**Comment by henrinie** [comment]
@jajokera @juzza81 You can take part in this too if you want to.

**Comment by ocrasborn** [comment]
Sure

**Comment by henrinie** [comment]
@ocrasborn What date and time would work for you? Anything between September 1st and 14th?
@codev @Woseseltops @vanlummelhuizen We are also waiting for your answer to this question! And if any of you know someone else we should include, please let me know.

**Comment by ocrasborn** [comment]
If we start at 12:00 Amsterdam time again, I could do 4-6-7-13-14 Sept.

**Comment by Woseseltops** [comment]
Assuming Amsterdam time 12:00h, I co do: 6 7 8 14. 

Overlap with Onno:
* 6 sept, 1000h GMT 
* 7 sept, 1000h GMT
* 14 sept, 1000h GMT

**Comment by henrinie** [comment]
Okay, lets try the times @Woseseltops suggested.

>  * 6 sept, 1000h GMT
>  * 7 sept, 1000h GMT
>  * 14 sept, 1000h GMT

@stevecassidy @ocrasborn @Woseseltops @vanlummelhuizen @susanodd @codev @juzza81 @jajokera @oukapipp 
Vote on doodle for the best date:
https://doodle.com/poll/tk48xsezw6st5u9m
**Please note the timezone!** Check whether you have daylight saving time (*summer time* in europe) or not on those dates.

**Comment by stevecassidy** [comment]
Done, I could make the 14th but would prefer not to since I’m flying off to Chile the next day and will likely be busy.

Thanks,

Steve

On 17 Aug 2017, 7:02 PM +1000, Henri <notifications@github.com>, wrote:

Okay, lets try the times @Woseseltops<https://github.com/woseseltops> suggested.

  *   6 sept, 1000h GMT
  *   7 sept, 1000h GMT
  *   14 sept, 1000h GMT

@stevecassidy<https://github.com/stevecassidy> @ocrasborn<https://github.com/ocrasborn> @Woseseltops<https://github.com/woseseltops> @vanlummelhuizen<https://github.com/vanlummelhuizen> @susanodd<https://github.com/susanodd> @codev<https://github.com/codev> @juzza81<https://github.com/juzza81> @jajokera<https://github.com/jajokera> @oukapipp<https://github.com/oukapipp>
Vote on doodle for the best date:
https://doodle.com/poll/tk48xsezw6st5u9m
Please note the timezone! Check whether you have daylight saving time (summer time in europe) or not on those dates.

**Comment by henrinie** [comment]
It seems that September 6th & 7th are our options.
Can anyone point this question to @vanlummelhuizen and @susanodd ?

**Comment by henrinie** [comment]
According to Doodle, Thursday, September 7 is okay for everyone who has answered the poll.
Therefore we will held the video conference on **September 7** starting at:
* **13:00 EEST**(GMT+3) Finland, 
* **12:00 CEST**(GMT+2) Holland, 
* **11:00 BST**(GMT+1) United Kingdom, 
* **20:00 AEST**(GMT+10) Australia East.
@stevecassidy Is that the correct time you expected? It seems that for last meeting we had your timezone as both, GMT+11 and GMT+10, according to messages in this thread.

**Comment by ocrasborn** [comment]
Thanks Henry for arranging this; see you then.

**Comment by stevecassidy** [comment]
Yes, we’re +10 just now, that’s fine with me. Look forward to meeting you all again.

Regards,

Steve

On 25 Aug 2017, 6:28 PM +1000, Henri <notifications@github.com>, wrote:

According to Doodle, Thursday, September 7 is okay for everyone who has answered the poll.
Therefore we will held the video conference on September 7 starting at:

  *   13:00 EEST(GMT+3) Finland,
  *   12:00 CEST(GMT+2) Holland,
  *   11:00 BST(GMT+1) United Kingdom,
  *   20:00 AEST(GMT+10) Australia East.
@stevecassidy<https://github.com/stevecassidy> Is that the correct time you expected? It seems that for last meeting we had your timezone as both, GMT+11 and GMT+10, according to messages in this thread.
